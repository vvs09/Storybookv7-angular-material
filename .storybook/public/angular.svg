<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
