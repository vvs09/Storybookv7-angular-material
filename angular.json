{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}