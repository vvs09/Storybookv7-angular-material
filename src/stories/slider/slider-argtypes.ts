export const sliderArgTypes = {
  color: {
    control: 'select',
    description: 'Palette color of the slider.',
    options: ['primary', 'accent', 'warn'],
    table: {
      category: 'Appearance',
      defaultValue: { summary: 'primary' },
      type: { summary: 'ThemePalette' },
    },
  },
  disableRipple: {
    control: 'boolean',
    description: 'Whether ripples are disabled in the slider.',
    table: {
      category: 'Behavior',
      defaultValue: { summary: 'false' },
      type: { summary: 'boolean' },
    },
  },
  disabled: {
    control: 'boolean',
    description: 'Whether the slider is disabled.',
    table: {
      category: 'Behavior',
      defaultValue: { summary: 'false' },
      type: { summary: 'boolean' },
    },
  },
  discrete: {
    control: 'boolean',
    description:
      'Whether the slider displays a numeric value label upon pressing the thumb.',
    table: {
      category: 'Appearance',
      defaultValue: { summary: 'false' },
      type: { summary: 'boolean' },
    },
  },
  displayWith: {
    control: 'text',
    description: 'Function to format the value displayed in the thumb label.',
    table: {
      category: 'Function',
      type: { summary: 'function' },
    },
  },
  max: {
    control: 'number',
    description: 'The maximum value that the slider can have.',
    table: {
      category: 'Limits',
      defaultValue: { summary: '100' },
      type: { summary: 'number' },
    },
  },
  min: {
    control: 'number',
    description: 'The minimum value that the slider can have.',
    table: {
      category: 'Limits',
      defaultValue: { summary: '0' },
      type: { summary: 'number' },
    },
  },
  showTickMarks: {
    control: 'boolean',
    description:
      'Whether the slider displays tick marks along the slider track.',
    table: {
      category: 'Appearance',
      defaultValue: { summary: 'false' },
      type: { summary: 'boolean' },
    },
  },
  step: {
    control: 'number',
    description: 'The values at which the thumb will snap.',
    table: {
      category: 'Behavior',
      defaultValue: { summary: '1' },
      type: { summary: 'number' },
    },
  },
  value: {
    control: 'number',
    description: 'The current value of the slider.',
    table: {
      category: 'Value',
      defaultValue: { summary: '0' },
      type: { summary: 'number' },
    },
  },
  dragEnd: {
    control: false,
    description: 'Event emitted when the slider thumb stops being dragged.',
    table: {
      category: 'Events',
      type: { summary: 'EventEmitter<MatSliderDragEvent>' },
    },
    action: 'dragEnd',
  },
  dragStart: {
    control: false,
    description: 'Event emitted when the slider thumb starts being dragged.',
    table: {
      category: 'Events',
      type: { summary: 'EventEmitter<MatSliderDragEvent>' },
    },
    action: 'dragStart',
  },
  valueChange: {
    control: false,
    description: 'Event emitted when the value is changed.',
    table: {
      category: 'Events',
      type: { summary: 'EventEmitter<number>' },
    },
    action: 'valueChange',
  },
  // Assuming 'percentage' is for documentation purposes and not an @Input or @Output
  percentage: {
    control: false,
    description: 'The percentage of the slider that coincides with the value.',
    table: {
      category: 'Information',
      type: { summary: 'number' },
    },
  },
};
