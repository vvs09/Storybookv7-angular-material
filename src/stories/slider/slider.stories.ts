import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import {
  MatSliderModule,
  MatSlider,
  MatSliderThumb,
} from '@angular/material/slider';
import { sliderArgTypes } from './slider-argtypes';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

const meta = {
  title: 'Components/Slider',
  component: MatSlider,
  tags: ['autodocs'],
  argTypes: sliderArgTypes,
  decorators: [
    moduleMetadata({
      imports: [
        MatSliderModule,
        MatSliderThumb,
        FormsModule,
        ReactiveFormsModule,
      ],
    }),
  ],
  render: (args:any) => ({
    props: args,
    styles: [
      `
        .example-label-container {
            display: flex;
            justify-content: space-between;
            margin: 20px 10px 0;
            max-width: 284px;
          }
          .example-result-card .example-value-label {
            font-weight: 600;
          }
          .example-margin {
            margin: 8px;
          }
        `,
    ],
    template: `
    <div class="example-label-container">
  </div>
    <mat-slider
    ${argsToTemplate(args)}
    class="example-margin">
  <input matSliderThumb >
</mat-slider>
   `,
  }),
};

export default meta;
type Story = StoryObj<MatSlider>;

export const Basic: Story = {
  args: {
    disabled: false,
    showTickMarks: false,
    discrete: true,
    max: 100,
    min: 0,
    step: 1,
    color: 'primary',
    disableRipple: false,
  },
};

export const ShowTickMarks: Story = {
  args: {
    ...Basic.args,
    min: 0,
    max: 1000,
    step: 1000,
    showTickMarks: true,
    discrete: false,
  },
};

export const Disabled: Story = {
  args: {
    ...Basic.args,
    disabled: true,
  },
};

export const Step: Story = {
  args: {
    ...Basic.args,
    step: 60,
  },
};

export const Discrete: Story = {
  args: {
    ...Basic.args,
    discrete: false,
  },
};

export const Accent: Story = {
  args: {
    ...Basic.args,
    color: 'accent',
  },
};

export const Warn: Story = {
  args: {
    ...Basic.args,
    color: 'warn',
  },
};

export const Limits: Story = {
  args: {
    ...Basic.args,
    min: 0,
    max: 1000,
  },
};
