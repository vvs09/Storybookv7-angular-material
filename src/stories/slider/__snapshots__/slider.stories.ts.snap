// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Slider Accent smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-accent mdc-slider--discrete"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="true"
              ng-reflect-max="100"
              ng-reflect-min="0"
              ng-reflect-step="1"
              ng-reflect-color="accent"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="1"
           min="0"
           max="100"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb mdc-slider__thumb--short-value"
                             ng-reflect-discrete="true"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text="0"
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__value-indicator-container">
        <div class="mdc-slider__value-indicator">
          <span class="mdc-slider__value-indicator-text">
            0
          </span>
        </div>
      </div>
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider Basic smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-primary mdc-slider--discrete"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="true"
              ng-reflect-max="100"
              ng-reflect-min="0"
              ng-reflect-step="1"
              ng-reflect-color="primary"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="1"
           min="0"
           max="100"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb mdc-slider__thumb--short-value"
                             ng-reflect-discrete="true"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text="0"
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__value-indicator-container">
        <div class="mdc-slider__value-indicator">
          <span class="mdc-slider__value-indicator-text">
            0
          </span>
        </div>
      </div>
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider Disabled smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-primary mdc-slider--disabled mdc-slider--discrete"
              ng-reflect-disabled="true"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="true"
              ng-reflect-max="100"
              ng-reflect-min="0"
              ng-reflect-step="1"
              ng-reflect-color="primary"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           disabled
           step="1"
           min="0"
           max="100"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb mdc-slider__thumb--short-value"
                             ng-reflect-discrete="true"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text="0"
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__value-indicator-container">
        <div class="mdc-slider__value-indicator">
          <span class="mdc-slider__value-indicator-text">
            0
          </span>
        </div>
      </div>
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider Discrete smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-primary"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="false"
              ng-reflect-max="100"
              ng-reflect-min="0"
              ng-reflect-step="1"
              ng-reflect-color="primary"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="1"
           min="0"
           max="100"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb"
                             ng-reflect-discrete="false"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider Limits smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-primary mdc-slider--discrete"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="true"
              ng-reflect-max="1000"
              ng-reflect-min="0"
              ng-reflect-step="1"
              ng-reflect-color="primary"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="1"
           min="0"
           max="1000"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb mdc-slider__thumb--short-value"
                             ng-reflect-discrete="true"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text="0"
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__value-indicator-container">
        <div class="mdc-slider__value-indicator">
          <span class="mdc-slider__value-indicator-text">
            0
          </span>
        </div>
      </div>
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider ShowTickMarks smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-primary mdc-slider--tick-marks"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="true"
              ng-reflect-discrete="false"
              ng-reflect-max="1000"
              ng-reflect-min="0"
              ng-reflect-step="1000"
              ng-reflect-color="primary"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="1000"
           min="0"
           max="1000"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
      <div class="mdc-slider__tick-marks">
        <div class="mdc-slider__tick-mark--inactive"
             style="transform: translateX(0px);"
        >
        </div>
        <div class="mdc-slider__tick-mark--inactive"
             style="transform: translateX(106px);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb"
                             ng-reflect-discrete="false"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider Step smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-primary mdc-slider--discrete"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="true"
              ng-reflect-max="100"
              ng-reflect-min="0"
              ng-reflect-step="60"
              ng-reflect-color="primary"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="60"
           min="0"
           max="100"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb mdc-slider__thumb--short-value"
                             ng-reflect-discrete="true"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text="0"
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__value-indicator-container">
        <div class="mdc-slider__value-indicator">
          <span class="mdc-slider__value-indicator-text">
            0
          </span>
        </div>
      </div>
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;

exports[`Components/Slider Warn smoke-test 1`] = `
<storybook-root _nghost-ng-c1837470598
                ng-version="17.2.2"
>
  <div _ngcontent-ng-c1837470598
       class="example-label-container"
  >
  </div>
  <mat-slider _ngcontent-ng-c1837470598
              class="mat-mdc-slider mdc-slider example-margin mat-warn mdc-slider--discrete"
              ng-reflect-disabled="false"
              ng-reflect-show-tick-marks="false"
              ng-reflect-discrete="true"
              ng-reflect-max="100"
              ng-reflect-min="0"
              ng-reflect-step="1"
              ng-reflect-color="warn"
              ng-reflect-disable-ripple="false"
  >
    <input _ngcontent-ng-c1837470598
           type="range"
           matsliderthumb
           class="mdc-slider__input"
           step="1"
           min="0"
           max="100"
           aria-valuetext="0"
           style="padding: 0px 16px; width: calc(100% + 10px); left: -21px;"
    >
    <div class="mdc-slider__track">
      <div class="mdc-slider__track--inactive">
      </div>
      <div class="mdc-slider__track--active">
        <div class="mdc-slider__track--active_fill"
             style="left: 0px; right: auto; transform-origin: left center; transform: scaleX(0.0267857);"
        >
        </div>
      </div>
    </div>
    <mat-slider-visual-thumb class="mdc-slider__thumb mat-mdc-slider-visual-thumb mdc-slider__thumb--short-value"
                             ng-reflect-discrete="true"
                             ng-reflect-thumb-position="2"
                             ng-reflect-value-indicator-text="0"
                             style="transform: translateX(3px);"
    >
      <div class="mdc-slider__value-indicator-container">
        <div class="mdc-slider__value-indicator">
          <span class="mdc-slider__value-indicator-text">
            0
          </span>
        </div>
      </div>
      <div class="mdc-slider__thumb-knob">
      </div>
      <div matripple
           class="mat-ripple mat-mdc-focus-indicator"
           ng-reflect-disabled="true"
      >
      </div>
    </mat-slider-visual-thumb>
  </mat-slider>
</storybook-root>
`;
